### PR TITLE
Amrex copy

### PR DESCRIPTION
**Pull request by atmyers**

### CODE REVIEW

**Comment by atmyers** [comment]
Edit - note, do not merge this yet. I want to make sure the tests all pass.

**Comment by atmyers** [comment]
Everything seems to be passing.

File: Src/Base/AMReX_GpuContainers.H [hunk 1]
```diff
@@ -52,16 +52,169 @@ namespace Gpu {
...
+    void copy (HostToDevice, InIter begin, InIter end, OutIter result) noexcept
     {
-#ifdef AMREX_USE_GPU
-        return thrust::copy(begin, end, result);
-#else
-        return std::copy(begin, end, result);
-#endif
+        using value_type = typename std::iterator_traits<InIter>::value_type;
+        auto size = std::distance(begin, end);
+        htod_memcpy(&(*result), &(*begin), size*sizeof(value_type));
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_GpuContainers.H]: Maybe we can add a static_assert making sure `vaule_type` is trivially copyable (or is thrust complex)?

**Comment by WeiqunZhang** [comment]
Looks good to me other than a minor comment.

**Pull request merged**